src/main/java/dansplugins/factionsystem/commands/abs/TabCompleterBase.java

```diff
@@ -0,0 +1,588 @@
+package dansplugins.factionsystem.commands.abs;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import dansplugins.factionsystem.data.PersistentData;
+import dansplugins.factionsystem.objects.domain.Faction;
+import dansplugins.factionsystem.services.ConfigService;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.TabCompleter;
+import org.bukkit.entity.Player;
+
+import javax.annotation.Nullable;
+
+public class TabCompleterBase implements TabCompleter {
+
+	PersistentData persistentData;
+	ConfigService configService;
+	public TabCompleterBase(PersistentData persistentData, ConfigService configService)
+	{
+		this.persistentData = persistentData;
+		this.configService = configService;
+	}
+	public  List<String> getOnlinePlayers(String partialName) {
+		return filterStartingWith(partialName, Bukkit.getOnlinePlayers().stream().map(OfflinePlayer::getName));
+	}
+
+	public String joinArgsBeyond(int index, String delim, String[] args) {
+		++index;
+		String[] data = new String[args.length - index];
+		System.arraycopy(args, index, data, 0, data.length);
+		return String.join(delim, data);
+	}
+
+	public List<String> filterStartingWith(String prefix, Stream<String> stream) {
+		return (List)stream.filter((s) -> {
+			return s != null && !s.isEmpty() && s.toLowerCase().startsWith(prefix.toLowerCase());
+		}).collect(Collectors.toList());
+	}
+
+	public List<String> filterStartingWith(String prefix, Collection<String> strings) {
+		return filterStartingWith(prefix, strings.stream());
+	}
+
+	public List<String> tackOnBeginningAndEndQuotes(List<String> targetedList)
+	{
+		final List<String> changed = new ArrayList<>();
+		for(String string : targetedList)
+		{
+			String changedString = '"' + string + '"';
+			changed.add(changedString);
+		}
+
+		return changed;
+	}
+
+	public String removeBeginningAndEndQuotes(String targetedString)
+	{
+		return targetedString.replace("\"", "");
+	}
+
+	List<String> argsLength1 = new ArrayList<String>();
+
+	@Override
+	public List<String> onTabComplete(CommandSender sender, Command command, String alias, String[] args)
```

Comment:
This function is _huge_. Perhaps we could break it down into smaller functions (e.g. for subcommands)?

Reply:
Seconded. Splitting this up into more methods would be good.

Reply:
If it works, though, we don't necessarily have to clean this up in this PR.